app/src/main/java/org/mozilla/fenix/wallpapers/WallpaperDownloader.kt

```diff
@@ -48,47 +48,29 @@ class WallpaperDownloader(
             Result.runCatching {
                 val response = client.fetch(request)
                 if (!response.isSuccess) {
-                    logger.error("Download response failure code: ${response.status}")
                     return@withContext
                 }
                 File(localFile.path.substringBeforeLast("/")).mkdirs()
                 response.body.useStream { input ->
                     input.copyTo(localFile.outputStream())
                 }
             }.onFailure {
+                // This should clean up any partial downloads
                 Result.runCatching {
                     if (localFile.exists()) {
                         localFile.delete()
                     }
-                }.onFailure { e ->
-                    logger.error("Failed to delete stale wallpaper bitmaps while downloading", e)
                 }
-
-                logger.error(it.message ?: "Download failed: no throwable message included.", it)
             }
         }
     }
 
     private data class WallpaperMetadata(val remotePath: String, val localPath: String)
 
-    private fun Wallpaper.toMetadata(context: Context): List<WallpaperMetadata> =
-        listOf("landscape", "portrait").flatMap { orientation ->
-            listOf("light", "dark").map { theme ->
-                val localPath = "wallpapers/$orientation/$theme/$name.png"
-                val remotePath = "${context.resolutionSegment()}/" +
-                    "$orientation/" +
-                    "$theme/" +
-                    "${collection.name}/" +
-                    "$name.png"
-                WallpaperMetadata(remotePath, localPath)
-            }
+    private fun Wallpaper.toMetadata(): List<WallpaperMetadata> =
+        listOf(Wallpaper.ImageType.Portrait, Wallpaper.ImageType.Landscape).map { orientation ->
+            val localPath = getLocalPath(orientation, this.name)
+            val remotePath = "${collection.name}/${this.name}/${orientation.lowercase()}.png"
+            WallpaperMetadata(remotePath, localPath)
         }
```

Comment:
pretty clever way to create the list

Reply:
TY!

Reply:
Hopefully not too clever, in the "this is hard to reason about" way

Reply:
I mean, it does take a few more seconds to understand _how_ it's iterating over the image types, but it's still some cool kotlin-fu